src/main/java/com/manywho/services/saml/factories/JwtServiceFactory.java

```diff
@@ -2,26 +2,27 @@
 
 import com.manywho.services.saml.security.SecurityConfiguration;
 import com.manywho.services.saml.services.JwtService;
-import org.glassfish.hk2.api.Factory;
 
 import javax.inject.Inject;
+import javax.inject.Provider;
 import javax.inject.Singleton;
 
-public class JwtServiceFactory implements Factory<JwtService> {
+public class JwtServiceFactory implements Provider<JwtService> {
 
     private SecurityConfiguration securityConfiguration;
 
     @Inject
     JwtServiceFactory(SecurityConfiguration securityConfiguration) {
         this.securityConfiguration = securityConfiguration;
     }
+
     @Singleton
     public JwtService provide() {
         return new JwtService(securityConfiguration.getSecret());
```

Comment:
This method shouldn't be needed anymore, but make sure you register this provider as a `Singleton` in the binder if you want to keep it the same as before